### ISSUE TITLE
Search box: browser compatibility issue

### ISSUE THREAD

**Issue by alisonjo315** [CU-CommunityApps/cwd_base#8]
(Note: does not apply to current CD Demo site _yet_ -- not until latest theme changes are applied, bc right now CD Demo uses the Search box code from cwd_ssit.)

The search box created by cwd_base doesn't work on newer versions of Firefox, at least on Windows (I haven't asked anyone to test on Mac) -- Nick and I tested it on the new FCS site -- you end up on Cornell Search instead of the site search (but it works fine in Chrome).  (Caveat: I didn't check older Firefox, I just know that newer Firefox clashes with a lot of JS things out there in the world, so I'm making an assumption -- could be wrong, could be older Firefox, too.)

(I'm not sure if it's always been like this or if it's from recent theme enhancements.)

So far, seen on Windows 10 + Firefox 68-69.

Examples:
https://another-fcs-main.pantheonsite.io/
https://globallearning.cornell.edu/

Search box on CD Demo (with cwd_ssit theme) does not have the issue. Search box on Wildlife also does not have the issue. Haven't checked other sites yet.

(Notes: Nick said he's 51% sure this issue came up on Space Management and they fixed it -- something to check.)

"FWIW"... the issue was fixed on the new FCS site (that is, site search and Cornell search are now working) when Alison removed `$(window).on("load", function () {` from `fcs.js` -- this line is not in the version of cwd_ssit that's on CD Demo at the moment:
https://dev-fcs-main.pantheonsite.io

_Remedy INC000002633711_

**Comment by alisonjo315** [comment]
Just, wondering/thinking -- isn't it weird to have all those parameters in the search URLs, other than `keys=` or `q=`? -- oh, or, maybe it's leftover from a time when we had sites using Cornell search, so the "sitesearch" value would tell Cornell search which site to search...?

(example: https://dev-cd-demo.pantheonsite.io/search/node?keys=dev&btnG=go&sitesearch=thissite)

**Comment by stacypendell** [comment]
The extra parameters are part of a search template that UnivComm provides to help small sites managed by non-developers have a working search. No reason to have them here, they're leftovers.

**Comment by alisonjo315** [comment]
Ah!  Thanks, Stacy!

**Comment by philwilliammee** [comment]
@alisonjo315  site search does not seems to be implemented in the base theme cwd.js. It is in the child theme. project.js. I have a ticket about this https://github.com/CU-CommunityApps/cwd_project/issues/2

**Comment by philwilliammee** [comment]
There are improvements that can be made to global_header.html.twig that can improve search, 
changing line 65 to:
<form id="cu-search-form" role="search" action="/search/node/" method="get">
will default the method to **get** so that than can be removed from javascript function. Also if js breaks it will search this site by default and not "https://www.cornell.edu/search" which probably should be "//www.cornell.edu/search"

**Comment by woodseowl** [comment]
@philwilliammee Why would we not just use "https://..." these days? 

(Also, did some of your comment get lost? I can't figure out what you are suggesting the line be changed to.)

**Comment by philwilliammee** [comment]
yeah it did get lost  this is what I put.
```html
<form id="cu-search-form" role="search" action="/search/node/" method="get">
```

the "//" is minor bit its best practice IMO it will default to whatever protocol it comes from since the cd-demo site should be https it will use that, it just helps so you don't get warnings in the request header about insecure content.

**Comment by alisonjo315** [comment]
Ok sooooo, my instinct is to replace the default search box markup that's in cwd_base/html.html.twig with `/search/node`, buuuut the safest, most compatible out-of-the-box thing probably is for it to stay as Cornell search, b/c in theory, you don't *have* to have Drupal search enabled....... ALTHOUGH, "search this site" won't actually work out of the box, in this situation.......

Nonetheless, I'm gonna close this issue b/c the browser compatibility issue is def in the child theme, not cwd_base.  We can, of course, open a new issue if we want to change the default search box markup that's in cwd_base (and potentially add search box JS to cwd_base, too -- it's all borderline-hinky right now, conceptually if not technologically, but, speaking only for myself, I don't have a clean solution in mind quite yet).

_**Closing in favor of https://github.com/CU-CommunityApps/cwd_project/issues/2**_

**Comment by woodseowl** [comment]
>> re: protocol -- I mean for one thing, when we're in site development, it's often not HTTPS, right?

Just for clarity: the URL for Cornell search page is HTTPS, regardless of whether the current page is HTTP or HTTPS. We wouldn't get errors going from an HTTP to HTTPS page.
